### ISSUE TITLE
webAPI to list all files in a directory

### ISSUE THREAD

**Issue by shubhank008** [haiwen/seafile#1042]
Is there a webAPI or python API to list all files in a directory ?
I am trying to create a turn around way to share links of multiple files at once as seahub only allows sharing link one file at a time or whole directory.

My idea was to first get all repo/libraries of user, then all directories of user and then all files in a/each directory of user, using loops. Ofcourse in actual implementation user will be able to select which library/directory to enter and which files to share,then I will just have to call shareLink API with those files in a loop.

So far I found the list all repo/Libraries and all directories in a library APIs, but cannot seem to find a API to list all files in a directory or library

**Comment by lins05** [comment]
This no api to recursively list all files/directories.

...and all directories in a library...


This api would return both subfolders and files for the given path (again,
not recursively).

You can take the python web api code for a reference.

https://github.com/haiwen/python-seafile/tree/d126e5428e0adf71bd2dc17860781bcd5da65073/seafileapi/files.py#L140

**Comment by shubhank008** [comment]
My bad my bad, actually there are APIs to list the libraries (http://manual.seafile.com/develop/web_api.html#list-libraries) and list files in a directory (http://manual.seafile.com/develop/web_api.html#list-directory-entries)

What I was looking for list of directories in a library which is not directly included in API, but I paid a bit more attention to listDirectoryEntries now and it lists both files and directories (with their types), so I just need to start from ./ and I will get all files and directories in a repo at root level and can proceed from there

Basically, I am trying to implement a share multiple files link functionality
